prusti-viper/src/encoder/encoder.rs

```diff
@@ -713,6 +713,29 @@ impl<'v, 'tcx> Encoder<'v, 'tcx> {
         self.intern_viper_identifier(full_name, short_name)
     }
 
+    pub fn encode_lifetime_name(&self, region: &ty::Region) -> String {
+        match region.kind() {
+            ty::ReEarlyBound(_) => {
+                unimplemented!("ReEarlyBound: {}", format!("{}", region));
+            },
+            ty::ReLateBound(_, _) => {
+                unimplemented!("ReLateBound: {}", format!("{}", region));
+            },
+            ty::ReFree(_) => {
+                unimplemented!("ReFree: {}", format!("{}", region));
+            },
+            ty::ReStatic=> String::from("lft_static"),
+            ty::ReVar(region_vid) => format!("lft_{}", region_vid.index()),
+            ty::RePlaceholder(_) => {
+                unimplemented!("RePlaceholder: {}", format!("{}", region));
+            },
+            ty::ReEmpty(_) => {
+                unimplemented!("ReEmpty: {}", format!("{}", region));
+            },
+            ty::ReErased => String::from("lft_erased"),
+        }
+    }
```

Comment:
Makes sense. Actually, you may also need to use a custom `RegionKind` instead of `String` because we may need to distinguish the static lifetime from others. Also, for erased lifetimes it would be cleaner to have an enum variant instead of a special name. However, let leave this change for latter once we hit the actual tests.